### PR TITLE
Closes #3747: Update Docs to Remove 'wantsKeys' References and Corrects Changelog [ci skip]

### LINKED ISSUE #3747: Don't pass 'wantsKeys=true` for non-key bearing scopes
Currently we always invoke `beginOAuthFlowAsync` with wantsKeys=true, which will result in a 500 FxA page if the requested scopes aren't key-bearing (e.g., if sync scope wasn't requested).

In practice this means that it's currently not possible to sign-in without configuring sync.

### PR DESCRIPTION
**Pull request by rocketsroger**
…ts Changelog [ci skip]

---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by rocketsroger** [comment]
> I don't think we should change historical parts of the API since they do currently exist in those versions (0.30.0). Let's just correct which version the new API was added and document that the breaking API change is that those parameters are no longer needed.
> 
> Requesting changes because this can lead to confusion if merged.

Agreed.  I shouldn't have changed the older change logs  thanks,

File: components/service/firefox-accounts/README.md [hunk 1]
```diff
@@ -221,7 +221,7 @@ Once the configuration is available and an account instance was created, the aut
 
 ```kotlin
 launch {
-    val url = account.beginOAuthFlow(scopes, wantsKeys).await()
+    val url = account.beginOAuthFlow(scopes).await()
```
Comment (jonalmeida) [inline components/service/firefox-accounts/README.md]: This reminds me that I need to audit the APIs in components I've worked on to see if those READMEs have changed. 😅

**Comment by jonalmeida** [comment]
> Already running a review

Okayy fineeee.

**Comment by rocketsroger** [comment]
> > Already running a review
> 
> Okayy fineeee.

Haha

**Pull request merged**